Add support for nested var names in parser

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #75124
* #75119
* #75108

These occur with freezing cc @Krovatkin

Differential Revision: [D35373998](https://our.internmc.facebook.com/intern/diff/D35373998)

Commit:
Add support for nested var names in parser

[ghstack-poisoned]

Commit:
Update on "Add support for nested var names in parser"


These occur with freezing cc @Krovatkin 

[ghstack-poisoned]

davidberard98 reviewed (approved):
I think one of the tests from the previous PR ended up in this one.

Otherwise, looks good!

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35373998).

Commit:
Update on "Add support for nested var names in parser"


These occur with freezing cc @Krovatkin

Differential Revision: [D35373998](https://our.internmc.facebook.com/intern/diff/D35373998)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35373998).
